Sources/Automerge/Document.swift

```diff
@@ -768,6 +768,23 @@ public final class Document: @unchecked Sendable {
         }
     }
 
+    /// Commit the auto-generated transaction with options.
+    ///
+    /// - Parameters:
+    ///   - message: An optional message to attach to the auto-committed change (if any).
+    ///   - timestamp: An optional timestamp to attach to the auto-committed change (if any).
+    ///
+    /// The `commitWith` function also compacts the memory footprint of an Automerge document and increments the
```

Comment:
I think this doc comment is no longer accurate. `commitWith` doesn't compact the memory in any way. It just commits any uncommitted changes. Furthermore, if there are no outstanding changes (i.e. you haven't actually written anything to the doc since the last time you committed, or saved, or received a sync message etc.) then the heads of the document will not change.